tests/lib/util/source-code-fixer.js

```diff
@@ -242,6 +270,221 @@ describe("SourceCodeFixer", function() {
 
         });
 
+        describe("BOM manipulations", function() {
+
+            it("should insert BOM with an insertion of '\uFEFF' at 0", function() {
+                var result = SourceCodeFixer.applyFixes(sourceCode, [ INSERT_BOM ]);
+                assert.equal(result.output, "\uFEFF" + TEST_CODE);
+                assert.isTrue(result.fixed);
+                assert.equal(result.messages.length, 0);
+            });
+
+            it("should insert BOM with an insertion of '\uFEFFfoobar' at 0", function() {
+                var result = SourceCodeFixer.applyFixes(sourceCode, [ INSERT_BOM_WITH_TEXT ]);
+                assert.equal(result.output, "\uFEFF// start\n" + TEST_CODE);
+                assert.isTrue(result.fixed);
+                assert.equal(result.messages.length, 0);
+            });
+
+            it("should remove BOM with a negative range", function() {
+                var result = SourceCodeFixer.applyFixes(sourceCode, [ REMOVE_BOM ]);
+                assert.equal(result.output, TEST_CODE);
+                assert.isTrue(result.fixed);
+                assert.equal(result.messages.length, 0);
+            });
+
+            it("should replace BOM with a negative range and 'foobar'", function() {
```

Comment:
Seems the test description doesn't match what actually happens here... (foobar vs // start)